### PR TITLE
Functional Tests : BO - Catalog - Products : Header

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : BO - Catalog - Products : Header
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: & Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/5725727152
| Fixed ticket?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

**Comment by nesrineabdmouleh** [comment]
UI tests after fix sanity errors => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/5743927702

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh for the help :) 

@PrestaShop/ui-tests-maintainers & @PrestaShop/committers It's time for review

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 1]
```diff
@@ -68,14 +101,57 @@ class DescriptionTab extends BOBasePage {
...
+    captionFr: string|undefined,
+  ): Promise<string|null> {
+    // Select the image
+    await page.locator(this.productImage).nth(numImage - 1).click();
+
+    if (useAsCoverImage) {
+      await this.setCheckedWithIcon(page, this.productImageDropZoneCover, useAsCoverImage);
+    }
+    if (captionEn) {
+      await page.click(this.productImageDropZoneBtnLang);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
      await page.locator(this.productImageDropZoneBtnLang).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 2]
```diff
@@ -68,14 +101,57 @@ class DescriptionTab extends BOBasePage {
...
+      await this.elementVisible(page, this.productImageDropZoneDropdown);
+
+      await page.click(this.productImageDropZoneDropdownItem('en'));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
      await page.locator(this.productImageDropZoneDropdownItem('en')).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 3]
```diff
@@ -68,14 +101,57 @@ class DescriptionTab extends BOBasePage {
...
+      await this.setValue(page, this.productImageDropZoneCaption, captionEn);
+    }
+    if (captionFr) {
+      await page.click(this.productImageDropZoneBtnLang);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
      await page.locator(this.productImageDropZoneBtnLang).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 4]
```diff
@@ -68,14 +101,57 @@ class DescriptionTab extends BOBasePage {
...
+      await this.elementVisible(page, this.productImageDropZoneDropdown);
+
+      await page.click(this.productImageDropZoneDropdownItem('fr'));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
      await page.locator(this.productImageDropZoneDropdownItem('fr')).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 5]
```diff
@@ -68,14 +101,57 @@ class DescriptionTab extends BOBasePage {
...
+      await this.setValue(page, this.productImageDropZoneCaption, captionFr);
+    }
+
+    await page.click(this.productImageDropZoneBtnSubmit);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
    await page.locator(this.productImageDropZoneBtnSubmit).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/index.ts [hunk 6]
```diff
@@ -267,6 +405,27 @@ class CreateProduct extends BOBasePage {
...
 
+  /**
+   * Change product type
+   * @param page {Page} Browser tab
+   * @param productType {string} Data to choose in product type
+   * @returns {Promise<string>}
+   */
+  async changeProductType(page: Page, productType: string): Promise<string> {
+    // Click on the type label
+    await page.click(this.productTypePreview);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/index.ts]:
```suggestion
    await page.locator(this.productTypePreview).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/index.ts [hunk 7]
```diff
@@ -163,11 +279,33 @@ class CreateProduct extends BOBasePage {
...
+   * Set product name
+   * @param page {Page} Browser tab
+   * @param name {string} Name of the product
+   * @param locale {string} Locale
+   * @returns {Promise<void>}
+   */
+  async setProductName(page: Page, name: string, locale: string = 'en'): Promise<void> {
+    await this.waitForSelectorAndClick(page, this.productNameLanguageButton);
+    await this.waitForSelectorAndClick(page, this.productNameLanguageDropdownItem(locale));
+    await page.$eval(this.productNameLanguageDropdownItem(locale), (el: HTMLElement) => el.click());
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/index.ts]: We should start to get rid of `$eval` as it's discouraged by playwright

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 8]
```diff
@@ -105,8 +181,14 @@ class DescriptionTab extends BOBasePage {
 
     await this.addProductImages(page, [productData.coverImage, productData.thumbImage]);
 
+    await page.click(this.productSummaryTabLocale('en'));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
    await page.locator(this.productSummaryTabLocale('en')).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 9]
```diff
@@ -105,8 +181,14 @@ class DescriptionTab extends BOBasePage {
...
+    await this.elementVisible(page, `${this.productSummaryTabLocale('en')}.active`);
     await this.setValueOnTinymceInput(page, this.productSummary, productData.summary);
+
+    await page.click(this.productDescriptionTabLocale('en'));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
    await page.locator(this.productDescriptionTabLocale('en')).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 10]
```diff
@@ -105,8 +181,14 @@ class DescriptionTab extends BOBasePage {
...
+    await this.elementVisible(page, `${this.productDescriptionTabLocale('en')}.active`);
     await this.setValueOnTinymceInput(page, this.productDescription, productData.description);
+    await page.click(this.productSummaryTabLocale('fr'));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]:
```suggestion
    await page.locator(this.productSummaryTabLocale('fr')).click();
```

File: tests/UI/pages/BO/catalog/productsV2/add/index.ts (hunk 7 above)
Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/index.ts]: I agree... but with @nesrineabdmouleh we find only this solution.

When `page.$eval` will be deprecated, we will remove it.

Comment (boubkerbribri) [reply tests/UI/pages/BO/catalog/productsV2/add/index.ts]: Check if this works too, if not, we can let `$eval`
```suggestion
    await page.evaluate(
    (selector) => (document.querySelector(selector) as HTMLElement).click(),
    this.productNameLanguageDropdownItem(locale), 
    );
```

Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/index.ts]: Pushed

**Comment by Progi1984** [comment]
@boubkerbribri It's good the CI is :green_circle:

**Review by boubkerbribri — DISMISSED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by Progi1984** [comment]
@PrestaShop/committers Hi, Can i have a review, plz ?

**Review by nicosomb — APPROVED** [review]
ok for the twig part !

**Comment by nicosomb** [comment]
Thank you @Progi1984 !

**Pull request merged**